src/pages/spawn/spawnHost/spawnHostModal/transformer.test.ts

```diff
@@ -0,0 +1,67 @@
+import { formToGql } from "./transformer";
+
+describe("repo data", () => {
+  it("correctly converts from a form to GQL", () => {
+    expect(
+      formToGql({
+        formData: data.formData,
+        publicKeys: [],
+        spawnTaskData: null,
+      })
+    ).toStrictEqual(data.mutationInput);
+  });
+  const data = {
+    formData: {
+      distro: {
+        value: "ubuntu1804-workstation",
+        isVirtualWorkstation: true,
+      },
+      region: "rofl-east",
+      publicKeySection: {
+        useExisting: false,
+        newPublicKey: "blah blahsart",
+        publicKeyNameDropdown:
+          "a loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong name",
+        savePublicKey: true,
+        newPublicKeyName: "a name woo",
+      },
+      userdataScriptSection: {
+        runUserdataScript: true,
+        userdataScript: "a user data script",
+      },
+      setupScriptSection: {
+        defineSetupScriptCheckbox: true,
+        setupScript: "setup!!!",
+      },
+      expirationDetails: {
+        noExpiration: false,
+        expiration: "Thu Dec 08 2022 14:52:51 GMT-0500 (Eastern Standard Time)",
+      },
+      homeVolumeDetails: {
+        selectExistingVolume: false,
+        volumeSize: 504,
+        volumeSelect: "",
+      },
+    },
+    mutationInput: {
+      isVirtualWorkStation: true,
```

Comment:
We could probably test some more inputs here like not using a virtual workstation, no expiration, etc.